forms-flow-api/src/formsflow_api/services/import_support.py

```diff
@@ -101,22 +132,59 @@ def validate_input_data(self, request):
                 raise BusinessException(BusinessErrorCode.INVALID_INPUT) from err
         return request_data, file
 
-    def valide_form_exists(self, form_json, tenant_key):
+    def validate_form_exists(self, form_json, tenant_key, validate_path_only=False):
         """Validate form already exists."""
         title = form_json.get("title")
         name = form_json.get("name")
         path = form_json.get("path")
         # Add 'tenantkey-' from 'path' and 'name'
         if current_app.config.get("MULTI_TENANCY_ENABLED"):
-            name = f"{tenant_key}-name"
+            if not validate_path_only:
+                name = f"{tenant_key}-name"
             path = f"{tenant_key}-path"
-        query_params = f"title={title}&name={name}&path={path}&select=title,path,name"
+
+        # Build query params based on validation type
+        if validate_path_only:
+            query_params = f"path={path}&select=title,path,name,_id"
+        else:
+            query_params = (
+                f"title={title}&name={name}&path={path}&select=title,path,name"
+            )
         current_app.logger.info(f"Validating form exists...{query_params}")
-        response = self.formio.get_form_search(
-            query_params, self.__get_formio_access_token()
+        response = self.get_form_by_query(query_params)
+        return response
+
+    def validate_form_title(self, form_json, mapper):
+        """Validate form tile in the form_process_mapper table."""
+        # Exclude the current mapper from the query
+        current_app.logger.info(f"Validation for form title...{form_json.get('title')}")
+        mappers = FormProcessMapper.find_forms_by_title(form_json.get("title"))
+        current_app.logger.debug(f"mappers matching the title- {mappers}")
+        # If no mappers are found for the import form title, it's an invalid case
+        if not mappers:

```

Comment:
Can you verify this ? In import file, the title could be different than what's in the mapper table.

Reply:
Updated code to make valid if the title differs from what's in the mapper table.